fili-core/src/test/groovy/com/yahoo/bard/webservice/web/util/CacheServiceSpec.groovy

```diff
@@ -0,0 +1,203 @@
+        json = new JsonNodeFactory().arrayNode()
+        groupByQuery = RequestUtils.buildGroupByQuery()
+        querySigningService = Mock(SegmentIntervalsHashIdGenerator)
+        querySigningService.getSegmentSetId(_) >> Optional.of(1234L)
+        containerRequestContext = Mock(ContainerRequestContext)
+        containerRequestContext.getHeaders() >> (["Bard-Testing": "###BYPASS###", "ClientId": "UI"] as
+                MultivaluedHashMap<String, String>)
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        requestContext = new RequestContext(containerRequestContext, true)
+        requestContext.isReadCache() >> true
+        MAPPER.valueToTree(groupByQuery) >> json
+    }
+
+    def "Test if request is cachable"() {
+        setup:
+        response.getResponseContext() >> context
+
+        expect:
+        cacheService.isCacheable(response) == expected
+
+        where:
+        expected | context
+        true     | [:]
+        false    | createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): nonEmptyIntervals])
+        false    | createResponseContext([(VOLATILE_INTERVALS_CONTEXT_KEY.name): nonEmptyIntervals])
+        true     | createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): new SimplifiedIntervalList()])
+        true     | createResponseContext([(VOLATILE_INTERVALS_CONTEXT_KEY.name): new SimplifiedIntervalList()])
+        false    | createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): nonEmptyIntervals, (VOLATILE_INTERVALS_CONTEXT_KEY.name): nonEmptyIntervals])
+        false    | createResponseContext([(MISSING_INTERVALS_CONTEXT_KEY.name): new SimplifiedIntervalList(), (VOLATILE_INTERVALS_CONTEXT_KEY.name): nonEmptyIntervals])
+    }
+
+    def "Test getting key from druid query to check match with cache"() {
+        setup:
+        String expectedResponse = """{"aggregations":[],"context":{},"dataSource":{"name":"dataSource","type":"table"},"dimensions":[],"granularity":{"period":"P1D","type":"period"},"intervals":[],"postAggregations":[],"queryType":"groupBy"}"""
+
+        expect:
+        cacheService.getKey(groupByQuery) == expectedResponse
+    }
+
+    def "Test handle request on cache hit responds to the request"() {
+        expect: "The count of fact query cache hit is 0"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 0
+
+        when: "A query runs with a valid cache hit"
+        String cachedValue = cacheService.readCache(requestContext, groupByQuery)
+
+        then: "Check the cache and return valid json"
+        1 * dataCache.get(_) >> new MemTupleDataCache.DataEntry<String>("key1", 1234L, "value")
+
+        then: "The cahed value is retrieved"
+        cachedValue == "value"
+
+        and: "The count of fact query cache hit is incremented by 1"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 1
+    }
+
+    def "Test handle request cache miss"() {
```

Comment:
Favor the if/then nomenclature without the 'test' name.
So here, perhaps on 'When fetching a cache not in the data cache, return null'

Reply:
This applies throughout the tests.